### PR TITLE
esp32s2: Do a full scan when channel/BSSID are not given

### PR DESCRIPTION
**Pull request by BennyE**
Whenever you connect to your Wi-Fi network / SSID just with the following code the first AP that matches the SSID will be chosen, even if there is another AP with better signal strength (RSSI).

wifi.radio.connect(mysecrets["ssid"], mysecrets["password"])

This commit will enhance the function to do a full scan (across all channels) whenever channel/BSSID are omitted.

### CODE REVIEW

File: ports/esp32s2/common-hal/wifi/Radio.c [hunk 1]
```diff
@@ -142,6 +142,11 @@ wifi_radio_error_t common_hal_wifi_radio_connect(wifi_radio_obj_t *self, uint8_t
     } else {
         config->sta.bssid_set = 0;
     }
+    // If channel is 0 (default/unset) and BSSID is not given, do a full scan instead of fast scan
+    // This will ensure that the best AP in range is chosen automatically
+    if ((config->sta.bssid_set == 0) && (config->sta.channel == 0)) {
+        config->sta.scan_method = WIFI_ALL_CHANNEL_SCAN;
+    }
```
Comment (tannewt) [inline ports/esp32s2/common-hal/wifi/Radio.c:149]: Please set the FAST method in an else. Otherwise you won't be able to switch between the two.

**Review by tannewt — CHANGES_REQUESTED** [review]
Awesome! One small thing but good otherwise!

File: ports/esp32s2/common-hal/wifi/Radio.c (hunk 1 above)
Comment (BennyE) [reply ports/esp32s2/common-hal/wifi/Radio.c:149]: I'll update the PR this evening, but WIFI_FAST_SCAN would be the default anyway (behaviour as before the change).

```suggestion
    } else {
        config->sta.scan_method = WIFI_FAST_SCAN;
    }
```

**Review by BennyE — COMMENTED** [review]
Adding requested changes.

**Review by tannewt — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**